src/main/java/Position.java

```diff
@@ -0,0 +1,48 @@
+import exceptions.InvalidPositionException;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+public class Position {
+    private final int row;
```

Comment:
Can a Position have negative indexes? It would be better not to use int.

Reply:
Invalid indexes accounted for in the constructor

Reply:
I realized that Java doesn't have unsigned int, lol. My fault